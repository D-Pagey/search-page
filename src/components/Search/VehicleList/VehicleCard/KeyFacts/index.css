.key-facts {
  border-bottom: 1px solid #e4e4e4;
  display: flex;
  justify-content: space-around;
}

.key-facts-list {
  margin: 5px 10px 5px 0;
  padding-left: 30px;
}

.key-facts-list-item {
  text-transform: capitalize;
}