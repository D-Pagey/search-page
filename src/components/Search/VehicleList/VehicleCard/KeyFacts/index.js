import React from 'react';
import { objectOf, any } from 'prop-types';

import './index.css';

export default function KeyFacts({ data }) {
  const { 
    year,
    fuel,
    mpg,
    body_information,
    number_seats_information,
    color,
    transmission,
    number_doors_information
   } = data;

  return (
    <div className="key-facts">
      <ul className="key-facts-list">
        <li className="key-facts-list-item">{year}</li>
        <li className="key-facts-list-item">{fuel}</li>
        <li className="key-facts-list-item">{mpg} MPG</li>
      </ul>
      <ul className="key-facts-list">
        <li className="key-facts-list-item">{body_information}</li>
        <li className="key-facts-list-item">{number_seats_information} Seats</li>
        <li className="key-facts-list-item">{color}</li>
      </ul>
      <ul className="key-facts-list">
        <li className="key-facts-list-item">{transmission}</li>
        <li className="key-facts-list-item">{number_doors_information} Doors</li>
      </ul>
    </div>
  );
}

KeyFacts.propTypes = {
  data: objectOf(any).isRequired,
};
