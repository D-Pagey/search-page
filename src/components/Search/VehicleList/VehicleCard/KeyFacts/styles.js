import styled from 'styled-components';

export const KeyFactsWrapper = styled.div`
  border-bottom: 1px solid #e4e4e4;
  display: flex;
  justify-content: space-around;
`;

export const KeyFactsList = styled.ul`
  margin: 5px 10px 5px 0;
  padding-left: 30px;
`;

export const KeyFactsItem = styled.li`
  text-transform: capitalize;
`;
