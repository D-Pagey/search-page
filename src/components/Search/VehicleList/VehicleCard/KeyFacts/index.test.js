import React from 'react';
import { shallow } from 'enzyme';
import KeyFacts from '.';

it('should render', () => {
  const wrapper = shallow(<KeyFacts data={{}} />);
  expect(wrapper).toMatchSnapshot();
});
