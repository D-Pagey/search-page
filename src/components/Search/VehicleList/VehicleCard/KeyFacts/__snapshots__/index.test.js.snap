// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="key-facts"
>
  <ul
    className="key-facts-list"
  >
    <li
      className="key-facts-list-item"
    />
    <li
      className="key-facts-list-item"
    />
    <li
      className="key-facts-list-item"
    >
       MPG
    </li>
  </ul>
  <ul
    className="key-facts-list"
  >
    <li
      className="key-facts-list-item"
    />
    <li
      className="key-facts-list-item"
    >
       Seats
    </li>
    <li
      className="key-facts-list-item"
    />
  </ul>
  <ul
    className="key-facts-list"
  >
    <li
      className="key-facts-list-item"
    />
    <li
      className="key-facts-list-item"
    >
       Doors
    </li>
  </ul>
</div>
`;
